Gjoranv/add default public metrics consumer

gjoranv opened the pull request:
FYI: @bratseth 

The name and metrics contained in the `default-public` metrics consumer, hereunder the metric set with the same name, are new public apis (names and contents, not the code itself). They are not set in stone yet, so this is just a heads up.

The metric set and consumer names are equal to make it more obvious for users to create a custom consumer with some metrics added along with all metrics given to the default consumer.
```
  <consumer id='my-consumer' />
    <metric-set id='default-public' />
    <metric ... />
```

URL:
`host:19092/metrics/v1/values?consumer=default-public` vs.
`host:19092/metrics/v1/values?consumer=my-consumer`

gjoranv committed:
Add 'consumer' parameter to the generic metrics handler.

gjoranv committed:
Throw if there are two metric sets with the same id.

gjoranv committed:
Remove unnecessary suppressWarnings

gjoranv committed:
Rename default -> vespa and add clarifying comment.

gjoranv committed:
Add default public metrics consumer, only added for self-hosted.

+ Improve test setup with hosted/self-hosted parameter.

gjoranv committed:
Add consumers to health packet for down services.

gjoranv committed:
Add TODO for content metrics

gjoranv committed:
Remove network metrics from the default-public consumer.

gjoranv commented:
FYI: @jobergum

hmusum commented on config-model/src/main/java/com/yahoo/vespa/model/admin/monitoring/DefaultPublicMetrics.java:
```diff
@@ -0,0 +1,69 @@
+        metrics.add(new Metric("degraded_queries.rate"));
+        metrics.add(new Metric("failed_queries.rate"));
+        metrics.add(new Metric("serverActiveThreads.average")); // TODO: Remove in Vespa 8?
+
```
Should we use other metrics instead of the ones we want to remove in Vespa 8 then? Or don't we have other alternaatives yet?

gjoranv replied:
Discussed with @jobergum, and we probably want to keep these metrics on 8. I removed the TODOs.

hmusum reviewed (approved):
One comment, LGTM, although I guess we will want to discuss the name "default-public".

gjoranv committed:
Rename the public default consumer and metric set to 'default'

- Keep the class/constant names, as they are only used in a public
  context, and not for Oath-internal Vespa with Yamas.

gjoranv committed:
Remove TODO comments to remove 'average' metrics.

gjoranv committed:
Add TODOs for filtering of extra (node) metrics.

gjoranv committed:
Improve unit tests for the http handler.
